pkg/csprofiles/csprofiles.go

```diff
@@ -36,7 +115,27 @@ func GenerateDecisionFromProfile(Profile *csconfig.ProfileCfg, Alert *models.Ale
 		}
 		/*some fields are populated from the reference object : duration, scope, type*/
 		decision.Duration = new(string)
-		*decision.Duration = *refDecision.Duration
+		if Profile.Cfg.DurationExpr != "" && Profile.RuntimeDurationExpr != nil {
+			duration, err := expr.Run(Profile.RuntimeDurationExpr, exprhelpers.GetExprEnv(map[string]interface{}{"Alert": Alert}))
+			if err != nil {
+				log.Warningf("failed to run duration_expr : %v", err)
```

Comment:
we created a custom logger above (xlog / clog) shouldn't we use it ? (and same for further logging in this function)

Reply:
Nope, I removed the custom logger from `GenerateDecisionFromProfile`. We need to create a new logger only when we evaluate each profile (as each profile has it own log level).